mina.netty/src/main/java/org/jboss/netty/channel/socket/nio/AbstractNioWorker.java

```diff
@@ -629,9 +626,13 @@ public void run() {
                     int interestOps = channel.getInternalInterestOps();
```

Comment:
I ran some tests, it seems that your suggested change is fixing the issue without requiring the changes I did. I suggest to close this PR without merging and opening a new one only with those changes.